src/shared/utilities/fileSystemWatcher.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Interface representing an object that can listen to a vscode.FileSystemWatcher
+ */
+export interface FileWatcherListener extends vscode.Disposable {
```

Comment:
I'm not seeing the value in this interface...
* if we make additional implementations, they may not use the same events (for example, maybe it listens to changes but not deletes)
* the methods surfaced are really private methods. They aren't intended to be accessed publicly.

Reply:
Following offline discussion, I was having a fundamental misunderstanding of how events are handled in VS Code. Making significant changes to this section.